### PR TITLE
Allow status items with error background color (fix #110214)

### LINKED ISSUE #110214: Allow status items with error background color
Using the theme color for status bar items might not meet accessibility requirements (contrast). Is there another color I should use or do we actually need another theme color for this.
capture

### PR DESCRIPTION
**Pull request by bpasero**
This PR fixes #110214

### CODE REVIEW

File: src/vs/vscode.proposed.d.ts [hunk 1]
```diff
@@ -2403,4 +2403,89 @@ declare module 'vscode' {
...
 	}
 	//#endregion
+
+	//#region Statusbar Item Background Color (https://github.com/microsoft/vscode/issues/110214)
+
+	/**
+	 * A status bar item is a status bar contribution that can
+	 * show text and icons and run a command on click.
+	 */
+	export interface StatusBarItem {
```
Comment (gjsjohnmurray) [inline src/vs/vscode.proposed.d.ts:2413]: In https://github.com/microsoft/vscode/pull/111237#discussion_r529721545 @jrieken advised not replicating the existing members of an interface when proposing adding a new member.

Comment (jrieken) [reply src/vs/vscode.proposed.d.ts:2413]: 👍

File: src/vs/workbench/api/common/extHostStatusBar.ts [hunk 2]
```diff
@@ -72,6 +77,14 @@ export class ExtHostStatusBarEntry implements vscode.StatusBarItem {
 		return this._color;
 	}
 
+	public get backgroundColor(): ThemeColor | undefined {
+		if (this._extension && !this._extension?.enableProposedApi) {
+			throw new Error(`[${this._extension.identifier.value}]: Proposed API is only available when running out of dev or with the following command line switch: --enable-proposed-api ${this._extension.identifier.value}`);
+		}
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostStatusBar.ts:83]: Use `checkProposedApiEnabled` instead

File: src/vs/workbench/api/common/extHostStatusBar.ts [hunk 3]
```diff
@@ -95,6 +108,19 @@ export class ExtHostStatusBarEntry implements vscode.StatusBarItem {
 		this.update();
 	}
 
+	public set backgroundColor(color: ThemeColor | undefined) {
+		if (this._extension && !this._extension?.enableProposedApi) {
+			throw new Error(`[${this._extension.identifier.value}]: Proposed API is only available when running out of dev or with the following command line switch: --enable-proposed-api ${this._extension.identifier.value}`);
+		}
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostStatusBar.ts:114]: same

File: src/vs/workbench/api/common/extHostStatusBar.ts [hunk 4]
```diff
@@ -95,6 +108,19 @@ export class ExtHostStatusBarEntry implements vscode.StatusBarItem {
...
+
+		if (color && !ExtHostStatusBarEntry.ALLOWED_BACKGROUND_COLORS.includes(color.id)) {
+			color = undefined;
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostStatusBar.ts:117]: `Set` over array?

**Review by aeschli — APPROVED** [review]

**Comment by bpasero** [comment]
@misolori I merged this to get it into our November release, feel free to tweak colors as you see fit. For now I set the background to `red`:

https://github.com/microsoft/vscode/blob/14d59bef9272a309da6c99aa28eee1a26a8bd530/src/vs/workbench/common/theme.ts#L398-L409

**Pull request merged**